### ISSUE TITLE
issues with telemetry streaming to splunk when using legacy format

### ISSUE THREAD

**Issue by boreal321** [F5Networks/f5-telemetry-streaming#30]
##### ISSUE TYPE

 - Bug Report

##### Telemetry Streaming BUILD/ VERSION

```
{
    "nodeVersion": "v4.8.0",
    "version": "1.7.0",
    "release": "1",
    "schemaCurrent": "1.7.0",
    "schemaMinimum": "0.9.0"
}
```

##### BIGIP VERSION

```
13.1.0.7
```

##### SUMMARY

I'm running into issues sending APM logs to Splunk using Telemetry Streaming. It's probably not specific to APM but that's what I've been using to test.
 
The docs over here mention that to get similar output as the Analytics iApp which works well with F5 Analytics Splunk app you need to set the format to legacy. I've followed those instructions and added the additional config to my declaration which is accepted without producing any errors. But, when logs are generated they're not sent to Splunk and I get the following error in /var/log/restnoded/restnoded.log

<img width="1438" alt="image001" src="https://user-images.githubusercontent.com/138104/70551571-bfa24e80-1b45-11ea-8575-8a5e586fa914.png">

Sorry for the image. Copy and paste doesn't seem to work for me in UDF.
 
When I use a declaration without the legacy config the logs are being sent with no errors. This is good but I was hoping to be able to use the F5 Analytics Splunk App for the dashboards.

##### STEPS TO REPRODUCE

To reproduce I use the following declaration. Once that's applied I log into my application via an APM logon form. The session logs should be sent to Splunk but instead I get the error above.

```
{
    "class": "Telemetry",
    "controls": {
        "class": "Controls",
        "logLevel": "info"
    },
    "My_System": {
        "class": "Telemetry_System",
        "systemPoller": {
            "interval": 60,
            "actions": [ {
                "setTag": {
                    "facility": "TorontoLab"
                },
                "locations": {
                    "system": true
                }
              } ]
        }
    },
    "My_Listener": {
        "class": "Telemetry_Listener",
        "port": 6514
    },
    "My_Consumer": {
        "class": "Telemetry_Consumer",
        "type": "Splunk",
        "host": "10.1.1.4",
        "protocol": "https",
        "port": 8088,
        "allowSelfSignedCert": true,
        "passphrase": {
            "cipherText": "{{splunk_token}}"
        },
        "format": "legacy"
    }
}
```

##### EXPECTED RESULTS

When I use the following declaration the logs are sent to Splunk and each session variable is sent as a separate event. No errors are produced.

```
{
    "class": "Telemetry",
    "controls": {
        "class": "Controls",
        "logLevel": "info"
    },
    "My_System": {
        "class": "Telemetry_System",
        "systemPoller": {
            "interval": 60
        }
    },
    "My_Listener": {
        "class": "Telemetry_Listener",
        "port": 6514
    },
    "My_Consumer": {
        "class": "Telemetry_Consumer",
        "type": "Splunk",
        "host": "10.1.1.4",
        "protocol": "https",
        "port": 8088,
        "allowSelfSignedCert": true,
        "passphrase": {
            "cipherText": "{{splunk_token}}"
        }
    }
}

```

Here is are the AS3 declaration for setting up the log publisher for Telemetry Streaming

```
{
    "class": "ADC",
    "schemaVersion": "3.10.0",
    "remark": "Example depicting creation of BIG-IP module log profiles",
    "Common": {
        "Shared": {
            "class": "Application",
            "template": "shared",
            "telemetry_local_rule": {
                "remark": "Only required when TS is a local listener",
                "class": "iRule",
                "iRule": "when CLIENT_ACCEPTED {\n  node 127.0.0.1 6514\n}"
            },
            "telemetry_local": {
                "remark": "Only required when TS is a local listener",
                "class": "Service_TCP",
                "virtualAddresses": [
                    "255.255.255.254"
                ],
                "virtualPort": 6514,
                "iRules": [
                    "telemetry_local_rule"
                ]
            },
            "telemetry": {
                "class": "Pool",
                "members": [
                    {
                        "enable": true,
                        "serverAddresses": [
                            "255.255.255.254"
                        ],
                        "servicePort": 6514
                    }
                ],
                "monitors": [
                    {
                        "bigip": "/Common/tcp"
                    }
                ]
            },
            "telemetry_hsl": {
                "class": "Log_Destination",
                "type": "remote-high-speed-log",
                "protocol": "tcp",
                "pool": {
                    "use": "telemetry"
                }
            },
            "telemetry_formatted": {
                "class": "Log_Destination",
                "type": "splunk",
                "forwardTo": {
                    "use": "telemetry_hsl"
                }
            },
            "telemetry_publisher": {
                "class": "Log_Publisher",
                "destinations": [
                    {
                        "use": "telemetry_formatted"
                    }
                ]
            }
        }
    }
}
```

**Comment by dstokesf5** [comment]
Thank you for reporting this, I have copied it into our internal product backlog as AUTOTOOL-1150.

**Comment by boreal321** [comment]
Thanks @dstokesf5. Have you been able to reproduce this?

Has anyone had success using legacy format?

**Comment by dstokesf5** [comment]
I have not yet reproduced the issue, but I also did not follow your steps. The provided stack trace has pointed out a part of the code that looks incorrect. We have used the legacy format in the past to test parity with the Analytics iApp, so it at least works sometimes.

**Comment by dstokesf5** [comment]
This was resolved as part of the 1.11.0 release.
